src/security/ldap/src/main/java/org/geoserver/security/ldap/LDAPRoleService.java

```diff
@@ -222,7 +222,7 @@ public void executeWithContext(DirContext ctx,
                         String dn = LDAPUtils
                                 .getLdapTemplateInContext(ctx, template)
                                 .searchForSingleEntry("", userFilter,
-                                        new String[] { username }).getDn()
+                                        new String[] { username }).getNameInNamespace()
                                 .toString();
```

Comment:
As I commented on the commit diff, this change actually has nothing to do with the bug. But in the auth.prov. the parameter {2} applies to the full DN while in the role service it applies to the dn without base. I wanted to make this consistent with each other. So {2} always means the same thing